src/gtasa/effects/custom/unsorted/WASDcheatsEffect.cpp

```diff
@@ -0,0 +1,280 @@
+
+        eWeaponType weapons[] = {WEAPON_CHAINSAW, WEAPON_KATANA, WEAPON_SPAS12};
+        for (CPed *ped : CPools::ms_pPedPool)
+        {
+            if (!ped) continue;
+            if (ped == player) continue;
+
+            Command<eScriptCommands::COMMAND_SET_CHAR_RELATIONSHIP> (
+                ped, 4, ePedType::PED_TYPE_PLAYER1);
+            auto weapon
+                = weapons[inst->Random (0, 100000) % std::size (weapons)];
+
+            int m = CPickups::ModelForWeapon (weapon);
+            CStreaming::RequestModel (m, 2);
+            CStreaming::LoadAllRequestedModels (false);
+            CStreaming::SetModelIsDeletable (m);
+
+            Command<eScriptCommands::COMMAND_GIVE_WEAPON_TO_CHAR> (ped, weapon,
+                                                                   1);
+            Command<eScriptCommands::COMMAND_SET_CURRENT_CHAR_WEAPON> (ped,
+                                                                       weapon);
+            Command<eScriptCommands::COMMAND_TASK_KILL_CHAR_ON_FOOT> (ped,
+                                                                      player);
+        }
+    },
+    [] (EffectInstance *inst)
+    {
+        auto *player = FindPlayerPed ();
+        if (!player) return;
+
+        auto p = player->GetPosition ();
+        p.x += inst->Random (-15.0f, 15.0f);
+        p.y += inst->Random (-15.0f, 15.0f);
+        int explostionType = 11;
+        if (inst->Random (1, 100000) % 2 == 0)
+        {
+            explostionType = 0;
+        }
+        Command<eScriptCommands::COMMAND_ADD_EXPLOSION> (p.x, p.y, p.z,
+                                                         explostionType);
+    }};
+
+std::string cheatString;
+} // namespace
+
+class WASDcheatsEffect : public EffectBase
+{
+    int       timer      = 0;
+    int       CHECK_TIME = 5000;
+    const int SUBSTR_LEN = 10;
+
+    bool isWASDtyped ();
+
+public:
+    void
+    OnStart (EffectInstance *inst) override
+    {
+        cheatString.reserve (32);
+        cheatString.clear ();
+
+        HOOK_ARGS (inst, Hook_AddToCheatString, void (char key), 0x438480);
+    }
+
+    static void
+    Hook_AddToCheatString (auto &&cb, char key)
+    {
+        cb ();
+        if (!CCutsceneMgr::ms_running)
+        {
+            cheatString.append (1, key);
+        }
+    }
+
+    void
+    OnTick (EffectInstance *inst) override
+    {
+        if (!GameUtil::IsPlayerSafe ()) return;
+
+        timer += int (GenericUtil::CalculateTick ());
+
+        if (timer >= CHECK_TIME)
+        {
+            if (isWASDtyped ())
+            {
+                Command<eScriptCommands::COMMAND_PRINT_HELP> ("CHEAT1");
+                int cheat = inst->Random (0, 100000) % customCheats.size ();
+                customCheats[cheat](inst);
+            }
+            cheatString.clear ();
+            timer -= CHECK_TIME;
+        }
+    }
+};
+
+DEFINE_EFFECT (WASDcheatsEffect, "effect_wasd_cheats", 0);
+
+bool
+WASDcheatsEffect::isWASDtyped ()

```

Comment:
You can actually move this method inside WASDCheatsEffect's `public:` part instead of having it outside like this

As in, under `OnTick`